File: quizzes/quizzesV4/quizzes10/13-problem.ts
export {};

type NestedArray = number | NestedArray[];

const flattenArray = (args: NestedArray[], newArr: number[] = []) => {
    for (const item of args) {
        if (Array.isArray(item)) {
            flattenArray(item, newArr);
        } else {
            newArr.push(item);
        }
    }

    return newArr;
};

const arr: NestedArray[] = [1, [2, 3, 4], [5, 6, [7, 8]], 9, 10];

console.log(JSON.stringify({ data: flattenArray(arr) }, null, 2));

File: bins5/13-problem.rb
# frozen_string_literal: true

require 'json'

def flatten_array(args, new_arr = [])
  args.each do |item|
    if item.is_a?(Array)
      flatten_array(item, new_arr)
    else
      new_arr << item
    end
  end

  new_arr
end

arr = [ 1, [ 2, 3, 4 ], [ 5, 6, [ 7, 8 ] ], 9, 10 ]

puts JSON.pretty_generate({ data: flatten_array(arr) })
